openlibrary/tests/solr/test_update_work.py

```diff
@@ -81,74 +109,114 @@ def test_edition_count(self):
+        work = make_work()
+        update_work.data_provider = FakeDataProvider([
+            work,
+            make_edition(work, isbn_10=["123456789X"])
+        ])
         d = build_data(work)
         assert d['isbn'] == ['123456789X', '9781234567897']
 
-        work = make_work(editions=[
-                    make_edition(isbn_10=["9781234567897"])])
+        update_work.data_provider = FakeDataProvider([
+            work,
+            make_edition(work, isbn_10=["9781234567897"])
+        ])
         d = build_data(work)
         assert d['isbn'] == ['123456789X', '9781234567897']
 
     def test_other_identifiers(self):
-        work = make_work(editions=[
-                    make_edition(oclc_numbers=["123"], lccn=["lccn-1", "lccn-2"]),
-                    make_edition(oclc_numbers=["234"], lccn=["lccn-2", "lccn-3"]),
+        work = make_work()
+        update_work.data_provider = FakeDataProvider([
+            work,
+            make_edition(work, oclc_numbers=["123"], lccn=["lccn-1", "lccn-2"]),
+            make_edition(work, oclc_numbers=["234"], lccn=["lccn-2", "lccn-3"]),
         ])
         d = build_data(work)
         assert sorted(d['oclc']) == ['123', '234']
         assert sorted(d['lccn']) == ['lccn-1', 'lccn-2', 'lccn-3']
 
     def test_identifiers(self):
-        work = make_work(editions=[
-                    make_edition(identifiers={"librarything": ["lt-1"]}),
-                    make_edition(identifiers={"librarything": ["lt-2"]})
+        work = make_work()
+        update_work.data_provider = FakeDataProvider([
+            work,
+            make_edition(work, identifiers={"librarything": ["lt-1"]}),
+            make_edition(work, identifiers={"librarything": ["lt-2"]})
         ])
         d = build_data(work)
         assert sorted(d['id_librarything']) == ['lt-1', 'lt-2']
 
     def test_ia_boxid(self):
-        e = make_edition()
-        w = make_work(editions=[e])
+        w = make_work()
+        update_work.data_provider = FakeDataProvider([w, make_edition(w)])
         d = build_data(w)
         assert 'ia_box_id' not in d
 
-        e = make_edition(ia_box_id='foo')
-        w = make_work(editions=[e])
+        w = make_work()
+        update_work.data_provider = FakeDataProvider([w, make_edition(w, ia_box_id='foo')])
         d = build_data(w)
         assert 'ia_box_id' in d
         assert d['ia_box_id'] == ['foo']
 
     def test_with_one_lending_edition(self):
-        e = make_edition(key="/books/OL1M", ocaid='foo00bar', _ia_meta={"collection":['lendinglibrary', 'americana']})
-        w = make_work(editions=[e])
+        w = make_work()
+        update_work.data_provider = FakeDataProvider([
+            w,
+            make_edition(w, key="/books/OL1M", ocaid='foo00bar',
+                         _ia_meta={"collection": ['lendinglibrary', 'americana']})
+        ])
         d = build_data(w)
         assert d['has_fulltext'] == True
```

Comment:
We can likely just `assert` without True, or assert `is True`

Reply:
I actually prefer the explicit == True. It more explicitly declares the intent (and avoids any ambiguity about whether truthy values are acceptable).